### PR TITLE
Fix #930: Lock tempesta.sh to avoid issues with concurrent execution.

### LINKED ISSUE #930: Crash on scheduler module unregistering
Reported by @ikoveshnikov for the last master.

```
[  420.691487] [tempesta] Initializing Tempesta FW kernel module...
[  420.702209] [tempesta] Registering new scheduler: hash
[  420.714592] [tempesta] Registering new scheduler: http
[  420.723548] [tempesta] Registering new scheduler: ratio
[  420.735828] [tempesta] Prepearing for the configuration processing.
[  420.737746] [tempesta] Configuration processing is completed.
[  420.755368] [tdb] Opened table /opt/tempesta/db/filter.tdb: size=16777216 rec_size=20 base=ffff97677c400000
[  421.060102] [tdb] Opened table /opt/tempesta/db/cache.tdb: size=268435456 rec_size=0 base=ffff97676c400000
[  421.063670] [tempesta] Open listen socket on: 0.0.0.0
[  421.080266] [tempesta] modules are started
[  421.253431] TCP: request_sock_TCP: Possible SYN flooding on port 80. Sending cookies.  Check SNMP counters.
[  431.286332] [tempesta] Stopping all modules...
[  431.464210] [tdb] Close table 'cache0.tdb'
[  431.487355] [tdb] Close table 'filter0.tdb'
[  431.488938] [tempesta] modules are stopped
[  431.504405] [tempesta] New configuration is cleaned.
[  431.507095] [tempesta] Prepearing for the configuration processing.
[  431.509871] [tempesta] Warning: cache_methods: duplicate method: 'GET'
[  431.511601] [tempesta] Configuration processing is completed.
[  431.526032] [tempesta] Un-registering scheduler: hash
[  431.531513] [tdb] Opened table /opt/tempesta/db/filter.tdb: size=16777216 rec_size=20 base=ffff97677c400000
[  431.567913] ------------[ cut here ]------------
[  431.568018] kernel BUG at /home/user/tempesta/tempesta/tempesta_fw/sched/tfw_sched_http.c:579!
[  431.568018] invalid opcode: 0000 [#1] SMP
[  431.568018] Modules linked in: tfw_sched_ratio(O) tfw_sched_http(O-) tempesta_fw(O) tempesta_db(O) tempesta_tls(O) crct10dif_pclmul crc32_pclmul iTCO_wdt iTCO_vendor_support snd_hda_codec_generic binfmt_misc ppdev ghash_clmulni_intel snd_hda_intel snd_hda_codec snd_hda_core snd_hwdep snd_pcm snd_timer snd soundcore sg evdev virtio_balloon virtio_console qxl ttm drm_kms_helper lpc_ich pcspkr mfd_core serio_raw drm shpchp parport_pc parport button ip_tables x_tables autofs4 ext4 crc16 jbd2 crc32c_generic fscrypto ecb mbcache sr_mod cdrom crc32c_intel virtio_blk virtio_net aesni_intel ahci libahci aes_x86_64 glue_helper lrw gf128mul ablk_helper cryptd sym53c8xx scsi_transport_spi libata ehci_pci psmouse uhci_hcd ehci_hcd i2c_i801 i2c_smbus usbcore usb_common scsi_mod virtio_pci virtio_ring virtio [last unloaded: tfw_sched_hash][  431.568018] CPU: 0 PID: 1933 Comm: rmmod Tainted: G           O    4.9.0-tempesta-amd64 #1 Debian 4.9.35-tfw4-1
[  431.568018] Hardware name: QEMU Standard PC (Q35 + ICH9, 2009), BIOS 1.11.0-20171110_100015-anatol 04/01/2014
[  431.634325] [tdb] Opened table /opt/tempesta/db/cache.tdb: size=268435456 rec_size=0 base=ffff97676c400000
[  431.568018] task: ffff97675937e580 task.stack: ffffb514c12dc000
[  431.568018] RIP: 0010:[<ffffffffc06907ef>]  [<ffffffffc06907ef>] cleanup_module+0x2f/0x840 [tfw_sched_http]
[  431.568018] RSP: 0018:ffffb514c12dfee8  EFLAGS: 00010282
[  431.568018] RAX: ffffffffc06907c0 RBX: ffffffffc06921c0 RCX: 0000000000000000
[  431.568018] RDX: ffffffffc06924a0 RSI: ffffffffc06924c8 RDI: ffffffffac052f80
[  431.568018] RBP: 00005617442901f0 R08: 0000000000000070 R09: 0000000000000008
[  431.568018] R10: 0000000000000000 R11: ffff97675937e580 R12: 0000000000000800
[  431.568018] R13: 0000000000000800 R14: 0000000000000000 R15: 00005617442901f0
[  431.568018] FS:  00007f8b072d0700(0000) GS:ffff97677fc00000(0000) knlGS:0000000000000000
[  431.568018] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[  431.568018] CR2: 0000561744298a68 CR3: 000000003ecfa000 CR4: 00000000003406f0
[  431.568018] Stack:
[  431.568018]  ffffffffab50156c 656863735f776674 0000707474685f64 ffffb514c12dff58
[  431.568018]  0000000000000002 ffff97675937e580 ffffffffab40326c 00ffb514c12dff58
[  431.568018]  1694f676e711fafa 00007ffd99374fa8 00005617442901f0 00007ffd99375170
[  431.568018] Call Trace:
[  431.568018]  [<ffffffffab50156c>] ? SyS_delete_module+0x18c/0x260
[  431.568018]  [<ffffffffab40326c>] ? exit_to_usermode_loop+0x8c/0xb0
[  431.568018]  [<ffffffffaba0e1fb>] ? system_call_fast_compare_end+0xc/0x9b
[  431.568018] Code: 00 48 83 3d 33 1d 00 00 00 75 20 48 83 ec 08 48 c7 c7 60 21 69 c0 e8 31 63 0f 00 48 c7 c7 00 21 69 c0 48 83 c4 08 e9 11 4d 0f 00 <0f> 0b 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
[  431.568018] RIP  [<ffffffffc06907ef>] cleanup_module+0x2f/0x840 [tfw_sched_http]
[  431.568018]  RSP <ffffb514c12dfee8>
```

### PR DESCRIPTION
**Pull request by aleksostapenko**

### CODE REVIEW

File: scripts/tempesta.sh [hunk 1]
```diff
@@ -19,6 +19,9 @@
 # this program; if not, write to the Free Software Foundation, Inc., 59
 # Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 
+[ "${TEMPESTA_LCK}" != "$0" ] && \
+    exec env TEMPESTA_LCK="$0" flock -n "/tmp/tempesta-lock-file" "$0" "$@"
```
Comment (krizhanovsky) [inline scripts/tempesta.sh]: Error message must be printed if the lock can not be acquired.

File: tempesta_fw/sched/tfw_sched_hash.c [hunk 2]
```diff
@@ -433,6 +433,15 @@ tfw_sched_hash_del_srv(TfwServer *srv)
 		call_rcu_bh(&cl->rcu, tfw_sched_hash_put_srv_data);
 }
 
+void
+tfw_sched_hash_refcnt(bool get)
+{
+	if (get)
+		try_module_get(THIS_MODULE);
```
Comment (ikoveshnikov) [inline tempesta_fw/sched/tfw_sched_hash.c]: This is very generic code, may be better to have it in `main.c`?

File: scripts/tempesta.sh (hunk 1 above)
Comment (aleksostapenko) [reply scripts/tempesta.sh]: Corrected.

File: tempesta_fw/sched/tfw_sched_hash.c (hunk 2 above)
Comment (krizhanovsky) [reply tempesta_fw/sched/tfw_sched_hash.c]: @ikoveshnikov  why do we need to get modules? I supposed that there is no way to unload a scheduler module if tempesta_fw.ko is loaded and references them...

Comment (ikoveshnikov) [reply tempesta_fw/sched/tfw_sched_hash.c]: `tempesta_fw` module is referenced by schedulers modules, not vice versa:
```
root@debian:/home/user/tempesta/tempesta# ./scripts/tempesta.sh --start
Starting Tempesta...
...enable RPS on lo
...enable RPS on enp3s0
...load Tempesta modules
Loading Tempesta kernel modules...
Loading module tempesta_tls 
Loading module tempesta_db 
Loading module tempesta_fw tfw_cfg_path=/home/user/tempesta/tempesta/etc/tempesta_fw.conf
Loading module tfw_sched_hash 
Loading module tfw_sched_http 
Loading module tfw_sched_ratio 
...start Tempesta FW
done
root@debian:/home/user/tempesta/tempesta# lsmod 
Module                  Size  Used by
tfw_sched_ratio        20480  0
tfw_sched_http         16384  0
tfw_sched_hash         16384  0
tempesta_fw          4734976  3 tfw_sched_ratio,tfw_sched_hash,tfw_sched_http
tempesta_db            40960  2 tempesta_fw,tfw_sched_hash
tempesta_tls          462848  1 tempesta_fw
...
root@debian:/home/user/tempesta/tempesta# rmmod tfw_sched_ratio
root@debian:/home/user/tempesta/tempesta# lsmod 
Module                  Size  Used by
tfw_sched_http         16384  0
tfw_sched_hash         16384  0
tempesta_fw          4734976  2 tfw_sched_hash,tfw_sched_http
tempesta_db            40960  2 tempesta_fw,tfw_sched_hash
tempesta_tls          462848  1 tempesta_fw
...
```
So manual module unloading is possible.

**Pull request merged**